contribute/design/1.x/upgrade/volume-pools-upgrade.md

```diff
@@ -0,0 +1,499 @@
+#indicated with VERIFY
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  #VERIFY that you have provided a unique name for this upgrade job.
+  #The name can be any valid K8s string for name. This example uses
+  #the following convention: jiva-vol-<flattened-from-to-versions>-<pv-name>
+  name: jiva-vol-100110-pvc-713e3bb6-afd2-11e9-8e79-42010a800065
+  #VERIFY the value of namespace is same as the namespace where openebs components
+  # are installed. You can verify using the command:
+  # `kubectl get pods -n <openebs-namespace> -l openebs.io/component-name=maya-apiserver`
+  # The above command should return status of the openebs-apiserver.
+  namespace: openebs
+spec:
+  backoffLimit: 4
+  template:
+    spec:
+      #VERIFY the value of serviceAccountName is pointing to service account
+      # created within openebs namespace. Use the non-default account.
+      # by running `kubectl get sa -n <openebs-namespace>`
+      serviceAccountName: openebs-maya-operator
+      containers:
+      - name:  upgrade
+        args: 
+        - "jiva-volume"
+        - "--from-version=1.0.0"
+        - "--to-version=1.1.0-RC2"
+        #VERIFY that you have provided the correct cStor PV Name
+        - "--pv-name=pvc-713e3bb6-afd2-11e9-8e79-42010a800065"
+        #Following are optional parameters
+        #Log Level
+        - "--v=4"
+        #DO NOT CHANGE BELOW PARAMETERS
+        env:
+        - name: OPENEBS_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        tty: true 
+        image: openebs/m-upgrade:dev-1000110RC2-072703
+      restartPolicy: Never
+---
+```
+
+Execute the Upgrade Job Spec
+```
+$ kubectl apply -f jiva-vol-100110-pvc713.yaml
+```
+
+You can check the status of the Job using commands like:
+```
+$ kubectl get job -n openebs
+$ kubectl get pods -n openebs #to check on the name for the job pod
+$ kubectl logs -n openebs jiva-upg-100111-pvc-713e3bb6-afd2-11e9-8e79-42010a800065-bgrhx
+```
+
+#### UpgradeTask CR Example
+
+Here is an example of UpgradeTask for upgrading Jiva Volume. The 
+conditions are added by the Upgrade Job
+
+```
+apiVersion: openebs.io/v1alpha1
+kind: UpgradeTask
+metadata:
+  name: upgrade-jiva-pv-001
+  namespace: openebs
+spec:
+  fromVersion: 0.9.0
+  toVersion: 1.0.0
+  #flags can be used to change the default behavior 
+  flags: 
+    #maximum seconds to wait at any given step in the upgrade
+    timeout: 0 #wait forever
+  #the resource represents the type of resource being upgraded. 
+  #some examples are jivaVolume, cstorVolume, cstorPool, cstorPoolClaim
+  #and so forth. Each resource may pass a different set of parameters
+  #to uniquely identify the resource being upgraded. 
+  resourceType: 
+    jivaVolume:
+      pvName: pvc-3d290e5f-7ada-11e9-b8a5-54e1ad4a9dd4
+      #flags can be used to change the default behaviour 
+      flags: 
+        #specify the steps that can be ignored on error.
```

Comment:
This is related to steps. However, the steps are specified in status. Can we rethink about steps & options w.r.t steps.